compiler/incremental-compilation-impl/src/org/jetbrains/kotlin/incremental/IncrementalCompilerRunner.kt

```diff
@@ -185,10 +182,10 @@ abstract class IncrementalCompilerRunner<
             }
 
             exitCode
-        } finally {
-            if (cachesMayBeCorrupted) {
-                cleanOutputsAndLocalStateOnRebuild(args)
-            }
+        } catch (e: Throwable) {
```

Comment:
@nav-nav: Hi Nataliya, last week @ausatiy mentioned that he'd like to remove this commit from this PR for "code style" reasons. However, I think this commit improves readability because:
  - The logic here is that if something is thrown, we'll clean up the state. The new code with `try-catch` directly matches this logic, whereas with the previous `try-finally` approach, the code is spread at 3 different places (line 118, 173, 188) and we'd have to follow the value of the `cachesMayBeCorrupted` variable (i.e., it takes extra effort to express the same thing).
   - The behaviors before and after this commit are exactly the same (note that `finally` catches `Throwable`, which includes `Error` and `Exception`). With this commit, it's easier to change `catch(e: Throwable)` to `catch(e: Exception)` or `catch(e: SomeSpecificException)` later if we want.

I'm fine if you want to remove this commit, but if so it would be nice if you could let me know your reasons so we can have a discussion here.